Code/GraphMol/TautomerQuery/Wrap/rdTautomerQuery.cpp

```diff
@@ -0,0 +1,145 @@
+  return HasSubstructMatch(target, self, recursionPossible, useChirality,
+                           useQueryQueryMatches);
+}
+
+PyObject *tautomerGetSubstructMatch(const TautomerQuery &self,
+                                    const ROMol &target,
+                                    bool useChirality = false,
+                                    bool useQueryQueryMatches = false) {
+  return GetSubstructMatch(target, self, useChirality, useQueryQueryMatches);
+}
+
+PyObject *tautomerGetSubstructMatches(const TautomerQuery &self,
+                                      const ROMol &target, bool uniquify = true,
+                                      bool useChirality = false,
+                                      bool useQueryQueryMatches = false,
+                                      unsigned int maxMatches = 1000) {
+  return GetSubstructMatches(target, self, uniquify, useChirality,
+                             useQueryQueryMatches, maxMatches);
+}
+
+PyObject *getTautomers(const TautomerQuery &self) {
+  auto tauts = self.getTautomers();
+  auto nTauts = tauts.size();
+  auto tuple = PyTuple_New(nTauts);
+  for (size_t i=0; i< nTauts; i++) {
+    PyTuple_SetItem(tuple, i, python::converter::shared_ptr_to_python(tauts[i]));
+  }
+  return tuple;
+}
+
+PyObject *tautomerGetSubstructMatchesWithTautomers(
+    const TautomerQuery &self, const ROMol &target, bool uniquify = true,
+    bool useChirality = false, bool useQueryQueryMatches = false,
+    unsigned int maxMatches = 1000) {
+  std::vector<MatchVectType> matches;
+  std::vector<ROMOL_SPTR> matchingTautomers;
+  SubstructMatchParameters params;
+  params.uniquify = uniquify;
+  params.useChirality = useChirality;
+  params.useQueryQueryMatches = useQueryQueryMatches;
+  params.maxMatches = maxMatches;
+
+  {
+    NOGIL gil;
+
+    SubstructMatchParameters matchParamters;
+    matches = self.substructOf(target, params, &matchingTautomers);
+  }
+  int const numberMatches = matches.size();
+  PyObject *res = PyTuple_New(numberMatches);
+  for (int idx = 0; idx < numberMatches; idx++) {
+    PyObject *pair = PyTuple_New(2);
+    PyTuple_SetItem(pair, 0, convertMatches(matches[idx]));
+    PyTuple_SetItem(
+        pair, 1,
+        python::converter::shared_ptr_to_python(matchingTautomers[idx]));
+    PyTuple_SetItem(res, idx, pair);
+  }
+  return res;
+}
+
+} // namespace
+
+struct TautomerQuery_wrapper {
+  static void wrap() {
+    RegisterVectorConverter<size_t>("UnsignedLong_Vect");
+    auto docString =
+        "The Tautomer Query Class.\n\
+  Creates a query that enables structure search accounting for matching of\n\
+  Tautomeric forms\n";
+
+    python::class_<TautomerQuery, boost::noncopyable>(
+        "TautomerQuery", docString, python::no_init)
+        .def("__init__", python::make_constructor(createDefaultTautomerQuery))
+        .def("__init__", python::make_constructor(&TautomerQuery::fromMol))
+        .def("IsSubstructOf", tautomerIsSubstructOf,
```

Comment:
I've added bindings to use `SubstructWatchParameters`. Since the keyword functions ware already present, I have followed the example of `MolBundle` and provided both flavors.